File: src/main/java/de/tectoast/emolga/commands/draft/PokeFansExportCommand.java
package de.tectoast.emolga.commands.draft;

import de.tectoast.emolga.commands.Command;
import de.tectoast.emolga.commands.CommandCategory;
import de.tectoast.emolga.commands.GuildCommandEvent;
import de.tectoast.emolga.utils.draft.Draft;
import org.jsolf.JSONArray;
import org.jsolf.JSONObject;

import java.util.ArrayList;
import java.util.Comparator;
import java.util.HashMap;

import static de.tectoast.emolga.bot.EmolgaMain.emolgajda;

public class PokeFansExportCommand extends Command {
    public PokeFansExportCommand() {
        super("pokefansexport", "Macht Pokefans Export lol", CommandCategory.Draft);
        setArgumentTemplate(ArgumentManagerTemplate.builder().add("draft", "Draft-Name", "Der Name der Liga, für die der Export gemacht werden soll", ArgumentManagerTemplate.draft())
                .setExample("!pokefansexport Emolga-Conference")
                .build());
        setCustomPermissions(PermissionPreset.CULT);
    }

    @Override
    public void process(GuildCommandEvent e) {
        JSONObject league = Draft.getLeagueStatic(e.getArguments().getText("draft"));
        JSONObject picksObj = league.getJSONObject("picks");
        JSONArray tosend = new JSONArray();
        ArrayList<String> ids = new ArrayList<>(picksObj.keySet());
        HashMap<String, String> names = new HashMap<>();
        emolgajda.getGuildById(league.getString("guild")).retrieveMembersByIds(ids.toArray(new String[0])).get().forEach(mem -> names.put(mem.getId(), mem.getEffectiveName()));
        for (String id : ids) {
            JSONArray picksArr = picksObj.getJSONArray(id);
            JSONArray oneUser = new JSONArray();
            oneUser.put(names.get(id).replaceAll("[^A-Za-z\\s]", ""));
            oneUser.put(e.getArg(0));
            JSONArray mons = new JSONArray();
            picksArr.toList().stream().map(o -> (String) ((HashMap<String, Object>) o).get("name"))
                    .sorted(Comparator.comparing(str -> getDataJSON().getJSONObject(getDataName((String) str)).getJSONObject("baseStats").getInt("spe")).reversed())
                    .map(str -> str
                            .replace("Boreos-T", "Boreos Tiergeistform")
                            .replace("Voltolos-T", "Voltolos Tiergeistform")
                            .replace("Demeteros-T", "Demeteros Tiergeistform")
                            .replace("Boreos-I", "Boreos Inkarnationsform")
                            .replace("Voltolos-I", "Voltolos Inkarnationsform")
                            .replace("Demeteros-I", "Demeteros Inkarnationsform")
                            .replace("Wolwerock-Tag", "Wolwerock Tagform")
                            .replace("Wolwerock-Nacht", "Wolwerock Nachtform")
                            .replace("Wolwerock-Zw", "Wolwerock Zwielichtform")
                            .replace("Shaymin", "Shaymin Landform")
                            .replace("Durengard", "Durengard Schildform")
                            .replace("Pumpdjinn", "Pumpdjinn XL")
                            .replace("M-", "Mega-")
                            .replace("A-", "Alola-")
                            .replace("G-", "Galar-")
                    ).forEach(mons::put);
            oneUser.put(mons);
            tosend.put(oneUser);
            if(tosend.toString().length() > 1500) {
                e.reply(tosend.toString());
                tosend.clear();
            }
        }
        e.reply(tosend.toString());
    }
}

File: src/main/kotlin/de/tectoast/emolga/commands/draft/PokeFansExportCommand.kt
package de.tectoast.emolga.commands.draft

import de.tectoast.emolga.bot.EmolgaMain
import de.tectoast.emolga.commands.*
import de.tectoast.emolga.utils.json.db
import kotlinx.serialization.encodeToString
import kotlinx.serialization.json.JsonArray
import kotlinx.serialization.json.JsonElement
import kotlinx.serialization.json.add
import kotlinx.serialization.json.buildJsonArray
import java.util.regex.Pattern

object PokeFansExportCommand : Command("pokefansexport", "Macht Pokefans Export lol", CommandCategory.Draft) {
    init {
        argumentTemplate = ArgumentManagerTemplate.builder().add(
            "draft",
            "Draft-Name",
            "Der Name der Liga, für die der Export gemacht werden soll",
            ArgumentManagerTemplate.draft()
        ).setExample("!pokefansexport Emolga-Conference").build()
        setCustomPermissions(PermissionPreset.CULT)
        slash()
    }

    override suspend fun process(e: GuildCommandEvent) {
        e.done(ephemeral = true)
        val tc = e.textChannel
        val league = db.league(e.arguments.getText("draft"))
        val picksObj = league.picks
        val tosend = mutableListOf<JsonElement>()
        val ids = ArrayList(picksObj.keys)
        val names = mutableMapOf<Long, String>()
        EmolgaMain.emolgajda.getGuildById(league.guild)!!.retrieveMembersByIds(ids).get()
            .forEach { names[it.idLong] = it.effectiveName }
        for (id in ids) {
            val picksArr = picksObj[id].names().map { it to getDataObject(it).speed }
            val oneUser = buildJsonArray {
                add(JUST_CHARS_AND_WHITESPACES.matcher(names[id]!!).replaceAll(""))
                add(e.getArg(0))
                val mons = buildJsonArray {
                    picksArr.asSequence().sortedByDescending {
                        it.second
                    }.map {
                        it.first.replace("-Therian", " Tiergeistform")
                            .replace("Wolwerock-Midnight", "Wolwerock Nachtform")
                            .replace("Wolwerock-Dusk", "Wolwerock Zwielichtform").replace("Shaymin", "Shaymin Landform")
                            .replace("Durengard", "Durengard Schildform").replace("Pumpdjinn", "Pumpdjinn XL")
                            .let { str ->
                                var tmp = str
                                for (suffix in listOf("Mega", "Alola", "Galar", "Hisui", "Paldea")) {
                                    if (tmp.endsWith("-$suffix")) tmp = suffix + "-" + tmp.substringBeforeLast("-")
                                }
                                tmp
                            }
                    }.forEach { add(it) }
                }
                add(mons)
            }
            tosend.add(oneUser)
        }
        tc.sendMessage("[").queue()
        tosend.chunked(2).forEach {
            tc.sendMessage(myJSON.encodeToString(JsonArray(it)).removeSurrounding("[", "]") + ",").queue()
        }
        tc.sendMessage("]").queue()
    }


    private val JUST_CHARS_AND_WHITESPACES = Pattern.compile("[^A-Za-z\\s]")

}
